src/WebJobs.Script.Grpc/MessageExtensions/StatusResultExtensions.cs

```diff
@@ -27,12 +28,30 @@ public static bool IsFailure(this StatusResult statusResult, out Exception excep
             }
         }
 
-        public static bool IsSuccess<T>(this StatusResult status, TaskCompletionSource<T> tcs)
```

Comment:
I don't think we need to rename these methods

Reply:
Okay. From my understanding there are 4 scenarios in which IsFailure is called, and only one is in the invocation code path. (and IsSuccess is only called in the invocation code path) Am I correct in assuming that we only want to throw an InvocationException in the invocation code path and keep RpcException for other scenarios?

Reply:
Hmm good point, let's review the other code paths and go from there